### PR TITLE
Improve TS documentation

### PR DESCRIPTION
**Pull request by Vadorequest**
## Status

Ready to be merged

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[x] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[x] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Updating TS doc mostly with (internal) variable names that make more sense for newcomers. No breaking change. Also added documentation related to ELK to soften the DX.

Issue Number: N/A

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -13,7 +13,8 @@
     "copy": "cp CNAME storybook-static/",
     "build-storybook": "build-storybook",
     "chromatic": "npx chromatic --project-token ugo6c655mvm --auto-accept-changes",
-    "link:reaflow": "yarn unlink && yarn link && cd node_modules/react && yarn unlink && yarn link && cd ../react-dom && yarn unlink && yarn link"
+    "link:reaflow": "yarn link && cd ./node_modules/react && yarn link && cd ../react-dom && yarn link",
```
Comment (amcdnl) [inline package.json:16]: What is this for?

Comment (Vadorequest) [reply package.json:16]: It's for collaborators of Reaflow only, it's to quickly link a custom project to the reaflow project.

This way, one can work on his own repo (e.g: a demo) using a local reaflow and iterate much more quickly.

Comment (amcdnl) [reply package.json:16]: I think given this is pretty specific to your workflow ( i handle this differently ) , I'd rather not include this in the core at this time.

Comment (Vadorequest) [reply package.json:16]: How do you do this then? Because using npm/yarn linking is very common for npm projects.

File: src/helpers/useUndo.ts [hunk 2]
```diff
@@ -178,7 +178,7 @@ export const useUndo = ({
       description: 'Undo changes',
       callback: (event) => {
         event.preventDefault();
-        if (!disabled) {
+        if (!disabled && canUndo) {
```
Comment (Vadorequest) [inline src/helpers/useUndo.ts:181]: @amcdnl Not sure about that, please confirm.

File: src/helpers/useUndo.ts [hunk 3]
```diff
@@ -190,7 +190,7 @@ export const useUndo = ({
       description: 'Redo changes',
       callback: (event) => {
         event.preventDefault();
-        if (!disabled) {
+        if (!disabled && canRedo) {
```
Comment (Vadorequest) [inline src/helpers/useUndo.ts:193]: @amcdnl Not sure about that, please confirm.

File: src/layout/elkLayout.ts [hunk 4]
```diff
@@ -5,24 +5,163 @@ import { formatText, measureText } from './utils';
...
+   */
   'elk.direction': 'DOWN',
+
+  /**
+   * XXX NOT SURE - Option doesn't seem to exist, should be "org.eclipse.elk.layered.layering.strategy"?
+   * Strategy for node layering.
+   *
+   * @see https://www.eclipse.org/elk/reference/options/org-eclipse-elk-layered-layering-strategy.html
+   */
   nodeLayering: 'INTERACTIVE',
```
Comment (Vadorequest) [inline src/layout/elkLayout.ts:74]: @amcdnl This option seems wrong and will cause a breaking change when updating elkjs.

See https://github.com/kieler/elkjs/issues/122#issuecomment-778803355

It should be renamed to `org.eclipse.elk.layered.layering.strategy`, if I understand it well.

File: src/Canvas.tsx [hunk 5]
```diff
@@ -44,6 +49,8 @@ export interface CanvasContainerProps extends CanvasProps {
 
   /**
    * Whether the canvas is pannable or not.
+   *
+   * TODO What does "pannable" means in Reaflow context? I don't understand it
    */
   pannable?: boolean;
```
Comment (Vadorequest) [inline src/Canvas.tsx:55]: @amcdnl I didn't quite get what this option do for real.

**Comment by Vadorequest** [comment]
@amcdnl Ready for your review

File: src/helpers/useUndo.ts (hunk 2 above)
Comment (amcdnl) [reply src/helpers/useUndo.ts:185]: Good catch!

File: src/helpers/useUndo.ts (hunk 3 above)
Comment (amcdnl) [reply src/helpers/useUndo.ts:197]: Good catch!

File: src/layout/elkLayout.ts (hunk 4 above)
Comment (amcdnl) [reply src/layout/elkLayout.ts:93]: Good w/ the change

File: package.json (hunk 1 above)
Comment (amcdnl) [reply package.json:16]: Let's remove this from core plz

File: src/Canvas.tsx [hunk 6]
```diff
@@ -44,6 +49,8 @@ export interface CanvasContainerProps extends CanvasProps {
 
   /**
    * Whether the canvas is pannable or not.
+   *
+   * TODO What does "pannable" means in Reaflow context? I don't understand it
```
Comment (amcdnl) [inline src/Canvas.tsx:53]: Remove comment - it means 'can pan' which is scrollbars. There is a demo of this turned off.

File: src/layout/elkLayout.ts [hunk 7]
```diff
@@ -5,24 +5,182 @@ import { formatText, measureText } from './utils';
...
+  /**
+   * Spacing to be preserved between pairs of connected components.
+   * This option is only relevant if ‘separateConnectedComponents’ is activated.
+   *
+   * @see https://www.eclipse.org/elk/reference/options/org-eclipse-elk-spacing-componentComponent.html
+   */
   'spacing.componentComponent': '70',
+
+  /**
+   * XXX NOT SURE - Should be spacing.baseValue?
```
Comment (amcdnl) [inline src/layout/elkLayout.ts:168]: Can you add a `TODO:` or `NOTE:` comment - editors will pick those up automatically.

**Review by amcdnl — CHANGES_REQUESTED** [review]
few minor things otherwise lgtm

File: src/layout/elkLayout.ts (hunk 7 above)
Comment (Vadorequest) [reply src/layout/elkLayout.ts:168]: You want to leave it there, and simply change it to a "TODO" if I understand you well?

Comment (amcdnl) [reply src/layout/elkLayout.ts:168]: Yes plz

File: package.json (hunk 1 above)
Comment (Vadorequest) [reply package.json]: I did but I still don't think it's the right choice. IMHO it'll make contributing harder, while it doesn't really cause any trouble for anyone not trying to contribute.

**Comment by Vadorequest** [comment]
@amcdnl Changes made, LGTM.

**Pull request merged**